python/samples/demos/guided_conversations/interactive_guided_conversation.py

```diff
@@ -0,0 +1,123 @@
+rules = [
+    "DO NOT write the poem for the student."
+    "Terminate the conversation immediately if the students asks for harmful or inappropriate content.",
+]
+
+# Conversation Flow (optional) - This defines in natural language the steps of the conversation.
+conversation_flow = """1. Start by explaining interactively what an acrostic poem is.
+2. Then give the following instructions for how to go ahead and write one:
+    1. Choose a word or phrase that will be the subject of your acrostic poem.
+    2. Write the letters of your chosen word or phrase vertically down the page.
+    3. Think of a word or phrase that starts with each letter of your chosen word or phrase.
+    4. Write these words or phrases next to the corresponding letters to create your acrostic poem.
+3. Then give the following example of a poem where the word or phrase is HAPPY:
+    Having fun with friends all day,
+    Awesome games that we all play.
+    Pizza parties on the weekend,
+    Puppies we bend down to tend,
+    Yelling yay when we win the game
+4. Finally have the student write their own acrostic poem using the word or phrase of their choice. Encourage them to be creative and have fun with it.
+After they write it, you should review it and give them feedback on what they did well and what they could improve on.
+Have them revise their poem based on your feedback and then review it again.
+"""
+
+# Context (optional) - This is any additional information or the circumstances the agent is in that it should be aware of.
+# It can also include the high level goal of the conversation if needed.
+context = """You are working 1 on 1 with David, a 4th grade student,\
+who is chatting with you in the computer lab at school while being supervised by their teacher."""
+
+
+# Resource Constraints (optional) - This defines the constraints on the conversation such as time or turns.
+# It can also help with pacing the conversation,
+# For example, here we have set an exact time limit of 25 minutes which the agent will try to fill.
+resource_constraint = ResourceConstraint(
+    quantity=10,
+    unit=ResourceConstraintUnit.TURNS,
+    mode=ResourceConstraintMode.EXACT,
+)
+
+
+async def main() -> None:
+    """Main function to interactively run a guided conversation.
+
+    The user can chat with this teaching agent until:
+    1. The user types 'exit' to end the conversation.
+    2. There's a KeyboardInterrupt or EOFError.
+    3. The conversation ends. This can be due to the agent ending the conversation, which will happen if the resource constraint is met, the artifact is complete, or the conversation just isn't making progress (user is not cooperative).
+    """
+
+    kernel = Kernel()
+    service_id = "gc_main"
+    kernel = add_service(kernel, service_id)
+
+    # Wrap these inputs into a GCInput object and instantiate the GuidedConversationAgent
+    guided_conversation_input = GCInput(
```

Comment:
is this GCInput thing needed, why not just pass those items into the GuidedConversation constructor?

Reply:
Resolved